pkg/pkg.bzl

```diff
@@ -428,37 +462,35 @@ pkg_deb_impl = rule(
             allow_files = True,
         ),
     },
+    provides = [PackageArtifactInfo],
 )
 
-def pkg_deb(name, package, archive_name = None, **kwargs):
+def pkg_deb(name, archive_name = None, **kwargs):
     """Creates a deb file. See pkg_deb_impl."""
-    archive_name = archive_name or name
-    version = kwargs.get("version") or ""
-    architecture = kwargs.get("architecture") or "all"
-    out_deb = "%s_%s_%s.deb" % (package, version, architecture)
-    out_changes = "%s_%s_%s.changes" % (package, version, architecture)
-
+    if archive_name:
+        # buildifier: disable=print
+        print("archive_name is deprecated. Use package_file_name or out to name the output.")
```

Comment:
Consider quoting attribute names here (e.g. `'archive_name'`) to ease readability.  This occurs a few places below (and maybe above, I might have missed some).

As an aside, Starlark really needs the ability to print out warnings and nonfatal errors with the appropriate ornamentation.  Being able to tag rule attributes as "deprecated" may also be a good feature.

Reply:
Done.
Agreed on the Stardoc changes.